### PR TITLE
Feat: initial implementation for single artifact release

### PR DESCRIPTION
**Pull request by hfend**
This PR introduces an initial working implementation for a Keygen.sh Release action.

### CODE REVIEW

File: __tests__/main.test.ts [hunk 1]
```diff
@@ -1,29 +1,6 @@
-import {wait} from '../src/wait'
-import * as process from 'process'
-import * as cp from 'child_process'
-import * as path from 'path'
 import {expect, test} from '@jest/globals'
 
-test('throws invalid number', async () => {
+test('dummy', async () => {
```
Comment (morgante) [inline __tests__/main.test.ts]: Tests would be nice!

File: src/api/index.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+  async artifactFileUpload(
+    signedS3UploadUrl: string,
+    file: Buffer
+  ): Promise<void> {
+    // NOTE:
+    // This endpoint expects the whole file to be loaded in memory.
+    // As a result, the upload operation is at the mercy of available Runner memory.
+    //
+    // We would've made the upload read the file as a stream (using fs.createReadStream()). However, S3 does not support this.
+    // We would've also considered supporting S3's Multipart upload (when a file is detected to be larger than a given threshold). Howeber, Keygen does not provide utilities for getting pre-signed part upload urls.
```
Comment (morgante) [inline src/api/index.ts]: What does S3 not support about it? Can't you send a streaming HTTP PUT regardless of what the API supports - this seems like a major limitation?

**Review by morgante — COMMENTED** [review]
Nice job! 🥳

File: src/api/index.ts (hunk 2 above)
Comment (hfend) [reply src/api/index.ts]: My bad, S3 supports streaming uploads just fine. Implemented in https://github.com/unhack/keygen-action/pull/11/commits/00251801f5a0eeb019bb7f1effbeac09d297e943

**Review by morgante — APPROVED** [review]

**Comment by morgante** [comment]
Looks like checks failed.

File: __tests__/main.test.ts (hunk 1 above)
Comment (hfend) [reply __tests__/main.test.ts]: Added a happy path test in https://github.com/unhack/keygen-action/pull/11/commits/ff9992ae9409113f3d7c8a0d5b90e98ad606184b

**Comment by hfend** [comment]
> Looks like checks failed.

Made fixes in https://github.com/unhack/keygen-action/pull/11/commits/61ff24f31a5ccdae0573bf8089104eedefa55b3b and https://github.com/unhack/keygen-action/pull/11/commits/460234213457f13354030944c6f570a0e3b4431b

**Review by morgante — APPROVED** [review]

**Pull request merged**